src/main/java/org/seqdoop/hadoop_bam/util/SAMOutputPreparer.java

```diff
@@ -58,7 +60,35 @@ public SAMOutputPreparer() {
 	public OutputStream prepareForRecords(
 			OutputStream out, final SAMFormat format,
 			final SAMFileHeader header)
-		throws IOException
+		throws IOException {
+
+        switch (format) {
+            case BAM:
+                prepareBAMStream(out, format, header);
```

Comment:
This should cover SAM too. There are checks for BAM in `prepareBAMStream`, and the method doc for `prepareForRecords` suggests that it's for SAM as it says "This includes writing the given SAM header and, in the case of BAM, writing some further metadata as well as compressing everything written."